tests/bsim/bluetooth/mesh/src/test_beacon.c

```diff
@@ -908,6 +971,539 @@ static void test_rx_secure_beacon_interval(void)
+}
+
+static void prov_and_conf_rx(uint16_t addr)
+{
+	provision(RX_ADDR);
+	common_configure(RX_ADDR);
+
+	target_node_alloc(rx_comp, TX_ADDR, dev_key);
+}
+
+static void test_priv_tx_init(void)
+{
+	provision(TX_ADDR);
+	common_configure(TX_ADDR);
+}
+
+static void wait_for_next_priv_beacon(uint8_t timeout)
+{
+	int err;
+
+	/* We expect beacon every BEACON_INTERVAL, give 1s tolerance for RX/TX sync */
+	err = k_sem_take(&prb_received, K_SECONDS(timeout + 1));
+	if (err) {
+		FAIL("Private Beacon not received");
+		return;
+	}
+
+	ASSERT_FALSE(bt_addr_le_eq(&curr_beacon_adv_addr, &last_beacon_adv_addr));
+
+	bt_addr_le_copy(&curr_beacon_adv_addr, &last_beacon_adv_addr);
+}
+
+static void common_rx_priv_setup(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &rx_comp);
+	prov_and_conf_rx(RX_ADDR);
+
+	k_sem_init(&prb_received, 0, 1);
+	bt_mesh_beacon_disable();
+
+	ASSERT_TRUE(!atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_INITIATOR));
+	ASSERT_TRUE(!atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS));
+	ASSERT_TRUE(!atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_PENDING));
+	ASSERT_TRUE(!atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_TEST));
+	ASSERT_TRUE(bt_mesh.iv_index == 0);
+
+	bt_le_scan_cb_register(&scan_callbacks);
+}
+
+static void test_tx_priv_on_iv_update(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &tx_comp);
+	test_priv_tx_init();
+
+	tx_on_iv_update_test();
+}
+
+static void common_test_priv_rx_ivu(void)
+{
+	/* shift beaconing time line to avoid boundary cases. */
+	k_sleep(K_SECONDS(1));
+
+	bt_mesh_iv_update_test(true);
+	ASSERT_TRUE(atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_TEST));
+
+	wait_for_next_priv_beacon(BEACON_INTERVAL);
+	ASSERT_TRUE(atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS));
+	ASSERT_TRUE(bt_mesh.iv_index == 1);
+
+	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+	memcpy(last_random, curr_random, 13);
+
+	wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+	ASSERT_TRUE(!atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS));
+	ASSERT_TRUE(bt_mesh.iv_index == 1);
+	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+
+	memcpy(last_random, curr_random, 13);
+
+	wait_for_next_priv_beacon(BEACON_INTERVAL);
+
+	ASSERT_TRUE(atomic_test_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS));
+	ASSERT_TRUE(bt_mesh.iv_index == 2);
+	ASSERT_TRUE(memcmp(last_random, curr_random, 13) != 0);
+}
+
+static void test_rx_priv_on_iv_update(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_priv_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 1;
+
+	err = bt_mesh_priv_beacon_cli_set(0, TX_ADDR, &val);
+	if (err) {
+		FAIL("Failed to enable Private Beacon (err=%d)", err);
+		return;
+	}
+
+	common_test_priv_rx_ivu();
+
+	PASS();
+}
+
+static void test_rx_priv_on_iv_update_long_int(void)
+{
+	struct bt_mesh_priv_beacon val;
+	int err;
+
+	common_rx_priv_setup();
+
+	val.enabled = 1;
+	val.rand_interval = 3;
```

Comment:
Only one line is diff from the previous function. Could you make a common function?